src/internal/kindenv/kindenv.go

```diff
@@ -0,0 +1,572 @@
+// Package kindenv manages Kind (github.com/kubernetes-sigs/kind) environments.
+package kindenv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/bazelbuild/rules_go/go/runfiles"
+	"github.com/docker/docker/api/types/network"
+	docker "github.com/docker/docker/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/backoff"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/log"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/kind/pkg/apis/config/v1alpha4"
+	"sigs.k8s.io/kind/pkg/cluster"
+)
+
+const (
+	// kindenvVersion is stored in the cluster and will inform developers that they should
+	// recreate their kind cluster when the stored version doesn't match this.
+	kindenvVersion = 1
+
+	// nodeImage is the kind node image to use.  If you change this, also change the version
+	// above.  Always include the sha256 checksum (due to quirks in Kind's release process,
+	// according to their docs).
+	nodeImage = "kindest/node:v1.29.0@sha256:eaa1450915475849a73a9227b8f201df25e55e268e5d619312131292e324d570"
```

Comment:
This should be a dependency in MODULE.bazel, but since Kind is kind of setup to find and push this image, I'm leaving it here for now.